lib/ansible/galaxy/collection/__init__.py

```diff
@@ -769,6 +769,19 @@ def install_collections(
                     "Skipping signature verification."
                 )
 
+            if concrete_coll_pin.type == 'galaxy':
+                signatures = concrete_coll_pin.src.get_collection_signatures(
+                    concrete_coll_pin.namespace, concrete_coll_pin.name, concrete_coll_pin.ver
+                )
```

Comment:
Looking at this, I think this also belongs in the `ansible.galaxy.dependency_resolution.dataclasses` module since it essentially calling internals.

Reply:
Could even be something like `concrete_coll_pin = concrete_coll_pin.with_signatures_repopulated()` that could encapsulate calling `get_collection_signatures()`, followed by the suggested below `with_signatures()`...

Reply:
Fixed (I think). I added a helper for Candidate. I didn't put it in the base class since the attribute is `signature_sources` rather than `signatures`.

Reply:
@s-hertel I think you may've misunderstood one aspect of what I suggested — I didn't mean for it to become a `@classmethod`, just a normal instance method that returns a new instance, initialized with the updated values.